src/libraries/System.Runtime/ref/System.Runtime.cs

```diff
@@ -4311,6 +4311,7 @@ public sealed partial class OperatingSystem : System.ICloneable, System.Runtime.
         public static bool IsAndroid() { throw null; }
         public static bool IsAndroidVersionAtLeast(int major, int minor = 0, int build = 0, int revision = 0) { throw null; }
         public static bool IsBrowser() { throw null; }
+        public static bool IsWASI() { throw null; }
```

Comment:
This will need to go through API review as it touches public API. I'd suggest commenting it out for now, if you need it somewhere you can use the `bool IsOSPlatform(string platform)` method.

Reply:
You can file an issue following https://github.com/dotnet/runtime/blob/main/docs/project/api-review-process.md for the API review. I think they'll prefer `IsWasi()` since that matches .NET naming guidelines.